Source/scrollrt.cpp

```diff
@@ -377,13 +356,74 @@ void DrawMonster(const Surface &out, int x, int y, int mx, int my, const Monster
 		return;
 	}
 
+	constexpr auto getMonsterModeDisplayName = [](MonsterMode monsterMode) {
+		switch (monsterMode) {
+		case MonsterMode::Stand:
+			return "standing";
+
+		case MonsterMode::MoveNorthwards:
+			return "moving (northwards)";
+
+		case MonsterMode::MoveSouthwards:
+			return "moving (southwards)";
+
+		case MonsterMode::MoveSideways:
+			return "moving (sideways)";
+
+		case MonsterMode::MeleeAttack:
+			return "attacking (melee)";
+
+		case MonsterMode::HitRecovery:
+			return "getting hit";
+
+		case MonsterMode::Death:
+			return "dying";
+
+		case MonsterMode::SpecialMeleeAttack:
+			return "attacking (special melee)";
+
+		case MonsterMode::FadeIn:
+			return "fading in";
+
+		case MonsterMode::FadeOut:
+			return "fading out";
+
+		case MonsterMode::RangedAttack:
+			return "attacking (ranged)";
+
+		case MonsterMode::SpecialStand:
+			return "standing (special)";
+
+		case MonsterMode::SpecialRangedAttack:
+			return "attacking (special ranged)";
+
+		case MonsterMode::Delay:
+			return "delaying";
+
+		case MonsterMode::Charge:
+			return "charging";
+
+		case MonsterMode::Petrified:
+			return "petrified";
+
+		case MonsterMode::Heal:
+			return "healing";
+
+		case MonsterMode::Talk:
+			return "talking";
+
+		default:
+			app_fatal("Invalid monster mode.");
+		}
+	};
+
 	int nCel = monster.AnimInfo.GetFrameToUseForRendering();
 	const auto *frameTable = reinterpret_cast<const uint32_t *>(monster.AnimInfo.pCelSprite->Data());
 	int frames = SDL_SwapLE32(frameTable[0]);
 	if (nCel < 1 || frames > 50 || nCel > frames) {
 		const char *szMode = "unknown action";
-		if (monster._mmode <= 17)
-			szMode = MonsterModeNames[monster._mmode];
+		if (static_cast<int>(monster._mmode) <= 17)
+			szMode = getMonsterModeDisplayName(monster._mmode);

```

Comment:
That way this check becomes unnecessary, szMode can be initialised directly from a call to `getMonsterModeDisplayName`

Reply:
It's not the same behavior, though. If we change the function to return something, we are now assigning it on all code paths.

I kept the behavior as before.

Reply:
The old backing type was unsigned, -1 would be treated as a large positive value and get excluded by this check. Casting to int here allows negative values to pass this check if `_mmode` is set to some garbage value or a marker value.

szMode is already assigned a value on every code path, using the existing default value as the default case in the switch doesn't change behaviour compared to the old implementation.

Reply:
I would be fine reducing this to :
`const char *szMode = getMonsterModeDisplayName(monster._mmode);`

Since we are already logging a bad animation state, potentially shutting down if the AI is also in a bad state seams reasonable. Funtime is likely over in either case.

This change is optional, though it get rid of a cast to int.